arrayfire_wrapper/_backend.py

```diff
@@ -238,3 +317,12 @@ def get_backend() -> Backend:
     """
 
     return __backend
+
+def set_backend(backend_type : BackendType) -> None:
```

Comment:
Let's choose our final approach for `set_backend`.
As the first option, I see the approach when we make backend public.
For example:
```
from aw import backend, BackendType
backend.set_backend(BackendType.cpu)
```

As for the second option, we can keep the backend private as it is now. In this case, I'd recommend renaming the class method to smth like `change_backend` and keeping the public function as it is - `set_backend`, 'coz it is intuitive for older users and we will keep the same API even after this whole refactoring. And, keeping different names we won't return to overcomplications in code